app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt

```diff
@@ -293,6 +295,38 @@ class QuestionPlayerActivityTest {
     }
   }
 
+  @Test
+  fun textQuestionPlayer_startQuestionPlayer_progressBarIsDisplayed() {
+    launchForSkillList(SKILL_ID_LIST).use {
+      onView(withId(R.id.question_progress_bar)).check(matches(isCompletelyDisplayed()))
+    }
+  }
+
+  @Test
+  fun textQuestionPlayer_openKeyboard_progressBarIsNotDisplayed() {
+    launchForSkillList(SKILL_ID_LIST).use {
+      selectMultipleChoiceOption(optionPosition = 2)
+      onView(withId(R.id.continue_navigation_button)).perform(click())
+
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.text_input_interaction_view)).perform(click())
```

Comment:
try putting a syn

```kotlin
  @Test
  fun textQuestionPlayer_openKeyboard_progressBarIsNotDisplayed() {
    launchForSkillList(SKILL_ID_LIST).use {
      selectMultipleChoiceOption(optionPosition = 2)
      onView(withId(R.id.continue_navigation_button)).perform(click())

      testCoroutineDispatchers.runCurrent()
      onView(withId(R.id.text_input_interaction_view)).perform(click())

      testCoroutineDispatchers.runCurrent()
      onView(withId(R.id.question_progress_bar)).check(matches(not(isDisplayed())))
    }
  }
```

Reply:
@FareesHussain Could you try this one this `testCoroutineDispatchers` and using it's different functions? If we can bring delay while the keyboard get's opened up.

Reply:
@anandwana001 It looks like keyboard is not supported in `Robolectric`

[Reference](http://robolectric.org/device-configuration/#:~:text=Keyboard%20availability)

Reply:
Can we ignore this for Robolectric?

Reply:
Yes, but we are not interacting with keyboard which is basically the `Keyboard availability` means I think, what we need here is just to wait till the keyboard get's open, so it seems like we need a bit of stable delay. 
Could you check if we can do this with any of the functions from `testCoroutineDispatchers` ?

Reply:
@anandwana001 I have already played a lot with `testCoroutineDispatchers` and `Thread.sleep()`. Delay doesn't seem to be a solution. What I doubt is that Roboelectric is not even able to open the keyboard. I have found some other similar hardware related issues in the code such as in `testProfileEdit_configChange_startWithUserProfile_clickDelete_checkOpensDeletionDialog` test where Roboelectic is not able to change the configuration to landscape (the test passed even when I removed the `delete profile` button from the landscape layout XML)
There seems to be some problem with Roboelectric itself.

Reply:
We have ongoing PR for fixing orientation change in robolectric, but I guess for now it's good to have the test written, we cannot confirm it will pass but it's good to have and when orientation change thing worked out, GitHub action will catch if anything fails.

Reply:
What I think here is when using robolectric it doesn't toggle the android keyboard

Reply:
Or the android keyboard is never displayed in Robolectric

Reply:
Adding this for reference - https://github.com/robolectric/robolectric/pull/4231

I think yes, we can mark it espresso-only test for now. @ArpitShukIa Please add a comment on a test about why we are keeping this test as espresso only. Also, add the link for reference `https://github.com/robolectric/robolectric/pull/4231`

Reply:
@anandwana001 Should I mark `textQuestionPlayer_openThenCloseKeyboard_progressBarIsDisplayed` as well to run only on Expresso. The test passes on Roboelectric just because the keyboard is never opened (and hence never closed).

Reply:
Yes, this seems like. a false passing. 
You can mark them to run on espresso only. 
`@RunOn(TestPlatform.ESPRESSO)`

Reply:
Done